### PR TITLE
add support for ExceptionWithContextInterface

### PR DESCRIPTION
**Pull request by zajca**
Exceptions can now have additional information's which are propagated to log and to response

### CODE REVIEW

File: tests/EventListener/ExceptionListenerTest.php [hunk 1]
```diff
@@ -52,14 +53,16 @@ public function testHandleException(): void
...
+        self::assertCount(5, $responseBody);
         self::assertStringStartsWith('exception-', $responseBody['exceptionId']);
         self::assertEquals('12', $responseBody['code']);
         self::assertEquals('Internal Server Error occurred.', $responseBody['error']);
         self::assertEquals('error', $responseBody['status']);
+        self::assertEquals(null, $responseBody['context']);
         $record = $handler->getRecords()[0];
         self::assertEquals(Logger::CRITICAL, $record['level']);
         self::assertEquals($exception, $record['context']['exception']);
+        self::assertEquals(null, $responseBody['context']['context']);
```
Comment (odinuv) [inline tests/EventListener/ExceptionListenerTest.php:65]:
```suggestion
        self::assertEquals(null,$$record['context']['context']);
```

Comment (zajca) [reply tests/EventListener/ExceptionListenerTest.php]: kůl na 7.4 to aj failnulo, o důvod víc proč sa bát upgradu na connection :D

**Review by odinuv — APPROVED** [review]
lgtm

File: tests/EventListener/ExceptionListenerTest.php (hunk 1 above)
Comment (odinuv) [reply tests/EventListener/ExceptionListenerTest.php]: jj, ten "Trying to access array offset on value of type null" uz nam parkrat zatopil

**Pull request merged**